EventsExpress/ClientApp/src/hoc/withAuthRedirect.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+const withAuthRedirect = (allowedRoles) =>
+    (Component) => {
+        class RedirectComponent extends React.Component {
+            render() {
+                if (!localStorage.getItem('token') && this.props.user.id === null) return <Redirect to='/unauthorized' />
+
+                if (localStorage.getItem('token') && this.props.user.id !== null && !allowedRoles.includes(this.props.user.role)) return <Redirect to='/forbidden' />
+
+                return <Component {...this.props} />
+            }
+        }
+
+        let mapStateToProps = (state) => ({
+            user: state.user
```

Comment:
Does this mapping work for page refresh now?
I would suggest using the jwt token for user roles detecting - it should be faster a lot.

Reply:
I'm using JWT for checking if user can be authorize and if yes then component  will be re rendered